src/Analyzers/Analyzers/test/StartupCSharpAnalyzerTest.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+
+namespace Microsoft.AspNetCore.Analyzers;
+
+internal sealed class StartupCSharpAnalyzerTest : CSharpAnalyzerTest<StartupAnalyzer, XUnitVerifier>
+{
+    public StartupCSharpAnalyzerTest(StartupAnalyzer analyzer, ImmutableArray<MetadataReference> metadataReferences)
+    {
+        StartupAnalyzer = analyzer;
+        SolutionTransforms.Add((solution, projectId) =>
+        {
+            var project = solution.GetProject(projectId);
+
+            return project
+                .WithCompilationOptions(((CSharpCompilationOptions)project.CompilationOptions)
+                    .WithOutputKind(OutputKind.WindowsApplication))
+             .WithMetadataReferences(metadataReferences)
```

Comment:
💡 This transform is unnecessary; you can add to `AdditionalMetadataReferences` property directly.

Reply:
I did not find a `AdditionalMetadataReferences` property on the type, but `TestState.AdditionalReferences` appears to work for this.